llvm/lib/CodeGen/InitUndef.cpp

```diff
@@ -152,7 +118,7 @@ bool RISCVInitUndef::handleReg(MachineInstr *MI) {
       continue;
     if (!UseMO.getReg().isVirtual())
       continue;
-    if (!isVectorRegClass(UseMO.getReg()))
+    if (!TII->isVectorRegClass(MRI->getRegClass(UseMO.getReg())))
```

Comment:
Actually, a second thought @ostannard, is this function needed? This makes sure there is a Pseudo instruction available but if one has not been implemented the issue will still exist for that instruction. 

I propose to remove the function and check all together, this will ensure that when adding support for new architectures, all the register classes that require support have the relevant pseudo instruction added.

Reply:
Yes, I think it would be better to delete this check, as the register class shouldn't change whether that transformation is necessary or not. My main concern here is that there's actually some other condition, and the original RISC-V implementation was using this as a proxy for it which happened to be correct for RISC-V, but not in general. Removing the condition and running the tests seems like a good way to check that.

Reply:
Check has been deleted. Tests pass locally so functionality is not lost and there are no new issues raised by removing the check.